transmission_interface/src/simple_transmission_loader.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright 2022 PAL Robotics S.L.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "transmission_interface/simple_transmission_loader.hpp"
+
+#include <rclcpp/logging.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+// Pluginlib
+#include <pluginlib/class_list_macros.hpp>
+
+#include <memory>
+
+// ros2_control
+#include "hardware_interface/component_parser.hpp"
+#include "hardware_interface/hardware_info.hpp"
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
+#include "transmission_interface/simple_transmission.hpp"
+#include "transmission_interface/transmission_interface_exception.hpp"
+
+namespace transmission_interface
+{
+std::shared_ptr<Transmission> SimpleTransmissionLoader::load(
+  const hardware_interface::TransmissionInfo & transmission_info)
+{
+  // Transmission instance
+  try
+  {
+    std::shared_ptr<Transmission> transmission(new SimpleTransmission(
+      transmission_info.joints[0].mechanical_reduction, transmission_info.joints[0].offset));
```

Comment:
isn't this the cause of the segfaults you see? you should check if `joints` is the right length, `[0]` will just segfault. Alternatively, you could just call `.at(0)` and leave the rest to exception handling:
https://www.cplusplus.com/reference/vector/vector/at/

Reply:
Okay that makes sense!

Reply:
Please, let's use checks. Having exception “flying” around is really not fun in complex systems.

Reply:
OK, so you are catching exceptions here, so it is OK, but you are not catching the case where index could be missing if using `.at(x)`.

Reply:
#642